mods/d2k/maps/atreides-01b/atreides01b.lua

```diff
@@ -0,0 +1,160 @@
+
+HarkonnenReinforcements = { }
+HarkonnenReinforcements["Easy"] =
+{
+	{ "rifle", "rifle" }
+}
+
+HarkonnenReinforcements["Normal"] =
+{
+	{ "rifle", "rifle" },
+	{ "rifle", "rifle", "rifle" },
+	{ "rifle", "trike" },
+}
+
+HarkonnenReinforcements["Hard"] =
+{
+	{ "rifle", "rifle" },
+	{ "trike", "trike" },
+	{ "rifle", "rifle", "rifle" },
+	{ "rifle", "trike" },
+	{ "trike", "trike" }
+}
+
+HarkonnenEntryWaypoints = { HarkonnenWaypoint1.Location, HarkonnenWaypoint2.Location, HarkonnenWaypoint3.Location, HarkonnenWaypoint4.Location }
+HarkonnenAttackDelay = DateTime.Seconds(30)
+HarkonnenAttackWaves = 1
+
+AtreidesReinforcements = { "rifle", "rifle", "rifle", "rifle" }
+AtreidesEntryPath = { AtreidesWaypoint.Location, AtreidesRally.Location }
+
+Messages =
+{
+	"Build a concrete foundation before placing your buildings.",
+	"Build a Wind Trap for power.",
+	"Build a Refinery to collect Spice.",
+	"Build a Silo to store additional Spice."
+}
+
+ToHarvest = 2500
+
+IdleHunt = function(actor)
+	if not actor.IsDead then
+		Trigger.OnIdle(actor, actor.Hunt)
+	end
+end
+
+Tick = function()
+	if HarkonnenArrived and harkonnen.HasNoRequiredUnits() then
+		player.MarkCompletedObjective(KillHarkonnen)
+	end
+
+	if player.Resources > ToHarvest - 1 then
+		player.MarkCompletedObjective(GatherSpice)
+	end
+
+	-- player has no Wind Trap
+	if (player.PowerProvided <= 20 or player.PowerState ~= "Normal") and DateTime.GameTime % DateTime.Seconds(32) == 0 then
+		HasPower = false
+		Media.DisplayMessage(Messages[2], "")
+	else
+		HasPower = true
+	end
+
+	-- player has no Refinery and no Silos
+	if HasPower and player.ResourceCapacity == 0 and DateTime.GameTime % DateTime.Seconds(32) == 0 then
+		Media.DisplayMessage(Messages[3], "")
+	end
+
+	if HasPower and player.Resources > player.ResourceCapacity * 0.8 and DateTime.GameTime % DateTime.Seconds(32) == 0 then
+		Media.DisplayMessage(Messages[4], "")
+	end
+
+	UserInterface.SetMissionText("Harvested resources: " .. player.Resources .. "/" .. ToHarvest, color)
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("Atreides")
+	harkonnen = Player.GetPlayer("Harkonnen")
+
+	color = HSLColor.New(170, 255, 200)
+
+	if Map.Difficulty == "Normal" then
+		HarkonnenAttackWaves = 5
+		ToHarvest = 3000
+	elseif Map.Difficulty == "Hard" then
+		HarkonnenAttackWaves = 12
+		ToHarvest = 3500
```

Comment:
You could put these values in a hash like you do for the reinforcements as well, if you wanted to.